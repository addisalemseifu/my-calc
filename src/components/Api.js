import { React, useState, useEffect } from 'react';
import '../index.css';

export default function Api() {
  const [data, setData] = useState([]);
  const [hasError, setHasError] = useState(false);
  const [isLoading, setIsLoading] = useState(false);
  const commentUrl = 'https://api.api-ninjas.com/v1/quotes?limit=3';

  useEffect(() => {
    const fetchData = async () => {
      setIsLoading(true);
      try {
        const quotes = await fetch(commentUrl, {
          method: 'GET',
          headers: { 'X-Api-Key': 'ST1XKraVMSM+a5l/yAfQ/w==qv9Hk7U7AjcuHlwZ' },
        });
        const quotesObject = await quotes.json();
        setData(quotesObject);
      } catch (error) {
        setHasError(true);
      }
      setIsLoading(false);
    };
    fetchData();
  }, [setData, setIsLoading]);

  if (hasError) return <div>This site is not responding at the moment!</div>;

  if (isLoading) return <div>Fetching data...</div>;

  return (
    <div className="my-quotes">
      {data.map((item) => (
        <h4 key={item.author}>
          -
          {' '}
          {item.quote}
        </h4>
      ))}
    </div>
  );
}
