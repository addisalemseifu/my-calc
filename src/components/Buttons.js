import React from 'react';
import '../index.css';

export default function Buttons({
  // eslint-disable-next-line react/prop-types
  symbol, className, handleClick,
}) {
  return (
    <button
      data-testid="calc-buttons"
      onClick={() => handleClick(symbol)}
      type="button"
      className={className}
    >
      {symbol}
    </button>
  );
}
