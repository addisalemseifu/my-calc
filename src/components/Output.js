import React from 'react';

export default function Output() {
  return (
    <div className='output'>0</div>
  )
}
