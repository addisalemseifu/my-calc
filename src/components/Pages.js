import React from 'react';
import '../index.css';
import { Link } from 'react-router-dom';

export default function Pages({ page, direction, className }) {
  return (
    <li className={className}><Link to={direction}>{page}</Link></li>
  );
}
