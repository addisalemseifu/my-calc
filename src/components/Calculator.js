import React from 'react';
import '../index.css';
import Output from './Output';

export default function Calculator() {
  return (
    <div className="calc-grid">
      <Output />
      <button className="clear" type="button">AC</button>
      <button type="button">+/-</button>
      <button type="button">%</button>
      <button className="operator" type="button">/</button>
      <button type="button">7</button>
      <button type="button">8</button>
      <button type="button">9</button>
      <button className="operator" type="button">x</button>
      <button type="button">4</button>
      <button type="button">5</button>
      <button type="button">6</button>
      <button className="operator" type="button">-</button>
      <button type="button">1</button>
      <button type="button">2</button>
      <button type="button">3</button>
      <button className="operator" type="button">+</button>
      <button className="zero" type="button">0</button>
      <button type="button">.</button>
      <button className="operator" type="button">=</button>
    </div>
  );
}
