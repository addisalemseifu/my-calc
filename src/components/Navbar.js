import React from 'react';
import Pages from './Pages';
import '../index.css';

export default function Navbar() {
  return (
    <div className="nav_bar">
      <h1 className="header">Math Magicians</h1>
      <ul className="links">
        <Pages className="my-home" page="Home" direction="/" />
        <Pages className="my-calc" page="Calculator" direction="/calculator" />
        <Pages className="my-quote" page="Quote" direction="/quote" />
      </ul>
    </div>
  );
}
