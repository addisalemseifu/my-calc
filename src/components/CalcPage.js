import React from 'react';
import Navbar from './Navbar';
import Calculator from './Calculator';

export default function CalcPage() {
  return (
    <div className="page">
      <Navbar />
      <div className="calc_container">
        <h2 className="sub_header">Lets do some math!</h2>
        <Calculator />
      </div>

    </div>
  );
}
