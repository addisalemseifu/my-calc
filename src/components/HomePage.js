import React from 'react';
import Navbar from './Navbar';
import '../index.css';

export default function HomePage() {
  const text = 'Lorem ipsum dolor sit amet consectetur adipisicing elit. Labore inventore distinctio obcaecati molestias nemo pariatur, illum provident qui voluptas harum officia dolorum asperiores itaque nobis velit! Necessitatibus eligendi accusamus quod voluptates consectetur, quaerat dolorem natus ut suscipit ad ratione, soluta doloremque! Rem minus autem labore quas omnis? Repellat voluptatem fugiat at, facilis officia consectetur quasi eum dolor, reprehenderit recusandae officiis quis nulla obcaecati praesentium cupiditate nihil ab! Quidem commodi eum doloremque. Iusto, officia? Et nam, eveniet provident voluptatem earum dolor?';
  return (
    <div className="page">
      <Navbar />
      <h2 className="sub_header">Welcome to our page!</h2>
      <p>
        {text}
        <br />
        <br />
        {text}
      </p>
    </div>
  );
}
