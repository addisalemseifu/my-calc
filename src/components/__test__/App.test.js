import React from 'react';
import {
  render, fireEvent, screen,
} from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
import { MemoryRouter, Router } from 'react-router-dom';
import userEvent from '@testing-library/user-event';
import '@testing-library/jest-dom';
import { createMemoryHistory } from 'history';

import CalcPage from '../CalcPage';
import HomePage from '../HomePage';
import Buttons from '../Buttons';
import Calculator from '../Calculator';
import Navbar from '../Navbar';
import Output from '../Output';
import Pages from '../Pages';
import Api from '../Api';
import App from '../../App';

describe('Output', () => {
  it('should matche DOM Snapshot', async () => {
    const tree = renderer.create(<MemoryRouter><Output /></MemoryRouter>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('render something', () => {
    const obj = {
      total: 30,
      next: 2,
      operation: '-',
    };
    render(<Output total={obj.total} next={obj.next} />);
    screen.debug();
    const totalValue = screen.getByTestId('total');
    const nextValue = screen.getByTestId('next');
    expect(totalValue.textContent).toBe('30');
    expect(nextValue.textContent).toBe('2');
  });
});

describe('Button', () => {
  it('should matche DOM Snapshot', async () => {
    const tree = renderer.create(<MemoryRouter><Buttons /></MemoryRouter>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('render button', () => {
    const addTotext = jest.fn();
    render(<Buttons type="button" symbol="+/-" handleClick={addTotext} />);
    screen.debug();
    const buttons = screen.getByTestId('calc-buttons');

    // interact with element
    fireEvent.click(buttons);

    expect(addTotext).toHaveBeenCalledTimes(1);
  });
});

describe('Calculator', () => {
  it('should matche DOM Snapshot', async () => {
    const tree = renderer.create(<MemoryRouter><Calculator /></MemoryRouter>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('render calculator', () => {
    render(<Calculator />);
    screen.debug();
    const buttons = screen.getAllByRole('button');
    expect(buttons.length).toBe(19);
    // interact with element
    expect(screen.getByText('5')).toBeInTheDocument();
    for (let i = 0; i < 10; i += 1) {
      expect(screen.getByText(`${i}`)).toBeInTheDocument();
    }

    expect(screen.getByText('-')).toBeInTheDocument();
    expect(screen.getByText('x')).toBeInTheDocument();
    expect(screen.getByText('+')).toBeInTheDocument();
    expect(screen.getByText('รท')).toBeInTheDocument();
    expect(screen.getByText('%')).toBeInTheDocument();
    expect(screen.getByText('+/-')).toBeInTheDocument();
    expect(screen.getByText('=')).toBeInTheDocument();
    expect(screen.getByText('AC')).toBeInTheDocument();
  });

  it('perform Multiplication', () => {
    render(<Calculator />);
    const output = screen.getByTestId('total');
    const next = screen.getByTestId('next');
    const numButton1 = screen.getByText('2');
    const numButton2 = screen.getByText('8');
    const multiplyOperation = screen.getByText('x');
    const equalOperation = screen.getByText('=');

    fireEvent.click(numButton1);
    expect(next.textContent).toBe('2');
    fireEvent.click(multiplyOperation);
    expect(next.textContent).toBe('');
    expect(output.textContent).toBe('2');
    fireEvent.click(numButton2);
    expect(next.textContent).toBe('8');
    fireEvent.click(equalOperation);
    expect(output.textContent).toBe('16');
  });

  it('perform Division', () => {
    render(<Calculator />);
    const output = screen.getByTestId('total');
    const next = screen.getByTestId('next');
    const numButton1 = screen.getByText('8');
    const numButton2 = screen.getByText('2');
    const divisionOperation = screen.getByText('รท');
    const equalOperation = screen.getByText('=');

    fireEvent.click(numButton1);
    expect(next.textContent).toBe('8');
    fireEvent.click(divisionOperation);
    expect(next.textContent).toBe('');
    expect(output.textContent).toBe('8');
    fireEvent.click(numButton2);
    expect(next.textContent).toBe('2');
    fireEvent.click(equalOperation);
    expect(output.textContent).toBe('4');
  });

  it('perform Addition', () => {
    render(<Calculator />);
    const output = screen.getByTestId('total');
    const next = screen.getByTestId('next');
    const numButton1 = screen.getByText('8');
    const numButton2 = screen.getByText('2');
    const additionOperation = screen.getByText('+');
    const equalOperation = screen.getByText('=');

    fireEvent.click(numButton1);
    expect(next.textContent).toBe('8');
    fireEvent.click(additionOperation);
    expect(next.textContent).toBe('');
    expect(output.textContent).toBe('8');
    fireEvent.click(numButton2);
    expect(next.textContent).toBe('2');
    fireEvent.click(equalOperation);
    expect(output.textContent).toBe('10');
  });
  it('perform Subtruction', () => {
    render(<Calculator />);
    const output = screen.getByTestId('total');
    const next = screen.getByTestId('next');
    const numButton1 = screen.getByText('8');
    const numButton2 = screen.getByText('2');
    const subtructOperation = screen.getByText('-');
    const equalOperation = screen.getByText('=');

    fireEvent.click(numButton1);
    expect(next.textContent).toBe('8');
    fireEvent.click(subtructOperation);
    expect(next.textContent).toBe('');
    expect(output.textContent).toBe('8');
    fireEvent.click(numButton2);
    expect(next.textContent).toBe('2');
    fireEvent.click(equalOperation);
    expect(output.textContent).toBe('6');
  });
});

describe('Navbar', () => {
  it('should matche DOM Snapshot', async () => {
    const tree = renderer.create(<MemoryRouter><Navbar /></MemoryRouter>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('render all the navbar features', () => {
    render(<MemoryRouter><Navbar /></MemoryRouter>);
    screen.debug();
    const mainHeader = screen.getByText('Math Magicians');
    const homeLink = screen.getByText('Home');
    const calculatorLink = screen.getByText('Calculator');
    const quoteLink = screen.getByText('Quote');

    expect(mainHeader).toBeInTheDocument();
    expect(homeLink).toBeInTheDocument();
    expect(calculatorLink).toBeInTheDocument();
    expect(quoteLink).toBeInTheDocument();
  });

  it('All the links are working', async () => {
    const history = createMemoryHistory({ initialEntries: ['/'] });
    render(
      <Router location={history.location} navigator={history}>
        <App />
      </Router>,
    );
    const linkElement1 = screen.getByRole('link', { name: /Home/i });
    const linkElement2 = screen.getByRole('link', { name: /Calculator/i });
    const linkElement3 = screen.getByRole('link', { name: /Quote/i });

    userEvent.click(linkElement1);
    expect(history.location.pathname).toBe('/');
    screen.debug();
    userEvent.click(linkElement2);
    expect(history.location.pathname).toBe('/calculator');
    userEvent.click(linkElement3);
    expect(history.location.pathname).toBe('/quote');
    screen.debug();
  });
});

describe('Pages', () => {
  it('should matche DOM Snapshot', async () => {
    const tree = renderer.create(<MemoryRouter><Pages /></MemoryRouter>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Api', () => {
  it('should matche DOM Snapshot', async () => {
    const tree = renderer.create(<MemoryRouter><Api /></MemoryRouter>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders a list of quotes', async () => {
    const data = [
      {
        quote: 'The will of man is his happiness.',
        author: 'Friedrich Schiller',
        category: 'happiness',
      },
      {
        quote: 'You shold find what you sout for.',
        author: 'Jhon',
        category: 'selfhelp',
      },
      {
        quote: 'The will of man is his happiness.',
        author: 'Schiller',
        category: 'fitness',
      },
    ];
    render(
      <div className="my-quotes">
        {data.map((item) => (
          <h3 title="qq" key={item.author}>
            -
            {' '}
            {item.quote}
          </h3>
        ))}
      </div>,
    );
    await screen.debug();
    const quotes = await screen.findAllByTitle('qq');

    expect(quotes).toHaveLength(3);
  });
});

describe('CalcPage', () => {
  it('should matche DOM Snapshot', async () => {
    const tree = renderer.create(<MemoryRouter><CalcPage /></MemoryRouter>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('HomePage', () => {
  it('should matche DOM Snapshot', async () => {
    const tree = renderer.create(<MemoryRouter><HomePage /></MemoryRouter>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('QuotePage', () => {
  it('should matche DOM Snapshot', async () => {
    const tree = renderer.create(<MemoryRouter><HomePage /></MemoryRouter>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
