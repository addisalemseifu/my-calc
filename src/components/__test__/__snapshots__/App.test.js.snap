// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Api should matche DOM Snapshot 1`] = `
<div
  className="my-quotes"
/>
`;

exports[`Button should matche DOM Snapshot 1`] = `
<button
  data-testid="calc-buttons"
  onClick={[Function]}
  type="button"
/>
`;

exports[`CalcPage should matche DOM Snapshot 1`] = `
<div
  className="page"
>
  <div
    className="nav_bar"
  >
    <h1
      className="header"
    >
      Math Magicians
    </h1>
    <ul
      className="links"
    >
      <li
        className="my-home"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li
        className="my-calc"
      >
        <a
          href="/calculator"
          onClick={[Function]}
        >
          Calculator
        </a>
      </li>
      <li
        className="my-quote"
      >
        <a
          href="/quote"
          onClick={[Function]}
        >
          Quote
        </a>
      </li>
    </ul>
  </div>
  <div
    className="calc_container"
  >
    <h2
      className="sub_header"
    >
      Lets do some math!
    </h2>
    <div
      className="calc-grid"
    >
      <div
        className="output"
      >
        <div
          className="result"
        >
          <h1
            data-testid="total"
          />
        </div>
        <div
          className="text"
        >
          <h3
            data-testid="next"
          />
        </div>
      </div>
      <button
        className="clear"
        data-testid="calc-buttons"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        data-testid="calc-buttons"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        data-testid="calc-buttons"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        data-testid="calc-buttons"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
      <button
        data-testid="calc-buttons"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        data-testid="calc-buttons"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        data-testid="calc-buttons"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="operator"
        data-testid="calc-buttons"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
      <button
        data-testid="calc-buttons"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        data-testid="calc-buttons"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        data-testid="calc-buttons"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="operator"
        data-testid="calc-buttons"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
      <button
        data-testid="calc-buttons"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        data-testid="calc-buttons"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        data-testid="calc-buttons"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="operator"
        data-testid="calc-buttons"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
      <button
        className="zero"
        data-testid="calc-buttons"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        data-testid="calc-buttons"
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="operator"
        data-testid="calc-buttons"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;

exports[`Calculator should matche DOM Snapshot 1`] = `
<div
  className="calc-grid"
>
  <div
    className="output"
  >
    <div
      className="result"
    >
      <h1
        data-testid="total"
      />
    </div>
    <div
      className="text"
    >
      <h3
        data-testid="next"
      />
    </div>
  </div>
  <button
    className="clear"
    data-testid="calc-buttons"
    onClick={[Function]}
    type="button"
  >
    AC
  </button>
  <button
    data-testid="calc-buttons"
    onClick={[Function]}
    type="button"
  >
    +/-
  </button>
  <button
    data-testid="calc-buttons"
    onClick={[Function]}
    type="button"
  >
    %
  </button>
  <button
    data-testid="calc-buttons"
    onClick={[Function]}
    type="button"
  >
    รท
  </button>
  <button
    data-testid="calc-buttons"
    onClick={[Function]}
    type="button"
  >
    7
  </button>
  <button
    data-testid="calc-buttons"
    onClick={[Function]}
    type="button"
  >
    8
  </button>
  <button
    data-testid="calc-buttons"
    onClick={[Function]}
    type="button"
  >
    9
  </button>
  <button
    className="operator"
    data-testid="calc-buttons"
    onClick={[Function]}
    type="button"
  >
    x
  </button>
  <button
    data-testid="calc-buttons"
    onClick={[Function]}
    type="button"
  >
    4
  </button>
  <button
    data-testid="calc-buttons"
    onClick={[Function]}
    type="button"
  >
    5
  </button>
  <button
    data-testid="calc-buttons"
    onClick={[Function]}
    type="button"
  >
    6
  </button>
  <button
    className="operator"
    data-testid="calc-buttons"
    onClick={[Function]}
    type="button"
  >
    -
  </button>
  <button
    data-testid="calc-buttons"
    onClick={[Function]}
    type="button"
  >
    1
  </button>
  <button
    data-testid="calc-buttons"
    onClick={[Function]}
    type="button"
  >
    2
  </button>
  <button
    data-testid="calc-buttons"
    onClick={[Function]}
    type="button"
  >
    3
  </button>
  <button
    className="operator"
    data-testid="calc-buttons"
    onClick={[Function]}
    type="button"
  >
    +
  </button>
  <button
    className="zero"
    data-testid="calc-buttons"
    onClick={[Function]}
    type="button"
  >
    0
  </button>
  <button
    data-testid="calc-buttons"
    onClick={[Function]}
    type="button"
  >
    .
  </button>
  <button
    className="operator"
    data-testid="calc-buttons"
    onClick={[Function]}
    type="button"
  >
    =
  </button>
</div>
`;

exports[`HomePage should matche DOM Snapshot 1`] = `
<div
  className="page"
>
  <div
    className="nav_bar"
  >
    <h1
      className="header"
    >
      Math Magicians
    </h1>
    <ul
      className="links"
    >
      <li
        className="my-home"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li
        className="my-calc"
      >
        <a
          href="/calculator"
          onClick={[Function]}
        >
          Calculator
        </a>
      </li>
      <li
        className="my-quote"
      >
        <a
          href="/quote"
          onClick={[Function]}
        >
          Quote
        </a>
      </li>
    </ul>
  </div>
  <h2
    className="sub_header"
  >
    Welcome to our page!
  </h2>
  <p>
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Labore inventore distinctio obcaecati molestias nemo pariatur, illum provident qui voluptas harum officia dolorum asperiores itaque nobis velit! Necessitatibus eligendi accusamus quod voluptates consectetur, quaerat dolorem natus ut suscipit ad ratione, soluta doloremque! Rem minus autem labore quas omnis? Repellat voluptatem fugiat at, facilis officia consectetur quasi eum dolor, reprehenderit recusandae officiis quis nulla obcaecati praesentium cupiditate nihil ab! Quidem commodi eum doloremque. Iusto, officia? Et nam, eveniet provident voluptatem earum dolor?
    <br />
    <br />
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Labore inventore distinctio obcaecati molestias nemo pariatur, illum provident qui voluptas harum officia dolorum asperiores itaque nobis velit! Necessitatibus eligendi accusamus quod voluptates consectetur, quaerat dolorem natus ut suscipit ad ratione, soluta doloremque! Rem minus autem labore quas omnis? Repellat voluptatem fugiat at, facilis officia consectetur quasi eum dolor, reprehenderit recusandae officiis quis nulla obcaecati praesentium cupiditate nihil ab! Quidem commodi eum doloremque. Iusto, officia? Et nam, eveniet provident voluptatem earum dolor?
  </p>
</div>
`;

exports[`Navbar should matche DOM Snapshot 1`] = `
<div
  className="nav_bar"
>
  <h1
    className="header"
  >
    Math Magicians
  </h1>
  <ul
    className="links"
  >
    <li
      className="my-home"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li
      className="my-calc"
    >
      <a
        href="/calculator"
        onClick={[Function]}
      >
        Calculator
      </a>
    </li>
    <li
      className="my-quote"
    >
      <a
        href="/quote"
        onClick={[Function]}
      >
        Quote
      </a>
    </li>
  </ul>
</div>
`;

exports[`Output should matche DOM Snapshot 1`] = `
<div
  className="output"
>
  <div
    className="result"
  >
    <h1
      data-testid="total"
    />
  </div>
  <div
    className="text"
  >
    <h3
      data-testid="next"
    />
  </div>
</div>
`;

exports[`Pages should matche DOM Snapshot 1`] = `
<li>
  <a
    href="/"
    onClick={[Function]}
  />
</li>
`;

exports[`QuotePage should matche DOM Snapshot 1`] = `
<div
  className="page"
>
  <div
    className="nav_bar"
  >
    <h1
      className="header"
    >
      Math Magicians
    </h1>
    <ul
      className="links"
    >
      <li
        className="my-home"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li
        className="my-calc"
      >
        <a
          href="/calculator"
          onClick={[Function]}
        >
          Calculator
        </a>
      </li>
      <li
        className="my-quote"
      >
        <a
          href="/quote"
          onClick={[Function]}
        >
          Quote
        </a>
      </li>
    </ul>
  </div>
  <h2
    className="sub_header"
  >
    Welcome to our page!
  </h2>
  <p>
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Labore inventore distinctio obcaecati molestias nemo pariatur, illum provident qui voluptas harum officia dolorum asperiores itaque nobis velit! Necessitatibus eligendi accusamus quod voluptates consectetur, quaerat dolorem natus ut suscipit ad ratione, soluta doloremque! Rem minus autem labore quas omnis? Repellat voluptatem fugiat at, facilis officia consectetur quasi eum dolor, reprehenderit recusandae officiis quis nulla obcaecati praesentium cupiditate nihil ab! Quidem commodi eum doloremque. Iusto, officia? Et nam, eveniet provident voluptatem earum dolor?
    <br />
    <br />
    Lorem ipsum dolor sit amet consectetur adipisicing elit. Labore inventore distinctio obcaecati molestias nemo pariatur, illum provident qui voluptas harum officia dolorum asperiores itaque nobis velit! Necessitatibus eligendi accusamus quod voluptates consectetur, quaerat dolorem natus ut suscipit ad ratione, soluta doloremque! Rem minus autem labore quas omnis? Repellat voluptatem fugiat at, facilis officia consectetur quasi eum dolor, reprehenderit recusandae officiis quis nulla obcaecati praesentium cupiditate nihil ab! Quidem commodi eum doloremque. Iusto, officia? Et nam, eveniet provident voluptatem earum dolor?
  </p>
</div>
`;
